### PR TITLE
Test Helper Changes

### PR DESCRIPTION
**Pull request by declandewet**
# Summary of Changes

- Updated dependencies
- Previously skipped tests now run on Travis
- Disabled `Roots.analytics` for test helpers
- You can now pass `{ async: true }` as the last parameter to every test helper method. This makes the method run asynchronously and return a promise. This however, does not apply to `project.compile` and `project.install_dependencies`, which both are now fully asynchronous by default, as opposed to a mix of sync and async logic
- Even though there is a breaking change, dependant code will not suffer from the change.

# Breaking Change

- [x] disable analytics for `project.compile` method
  - **why?** because it doesn't make sense for test helpers and causes inconsistent failures for `roots-contentful` on Travis
  - > /cc @jenius @kylemac I need a review on this. I wasn't sure if I should add an option to Roots that allows disabling analytics at the instance-level so it doesn't override global config (e.g. `new Roots(analytics: false)` but I figured this would be okay since it is for testing only. It only becomes an issue if two instances of Roots are running at the same time on the same machine and one of them is running tests, which I figured is super edge-case, also - any recommendations on how to test this piece of logic?

> **note:** even though this is a breaking change, it theoretically should not intefere with any dependent code. It simply removes an unwanted side effect of `RootsUtil.Helpers`. In other words, dependendant libraries need not worry about updating anything other than the version for `roots-util` as if there were no breaking changes.

# Non-Breaking Changes

- [x] swap out deprecated `fs.exists` methods for one that uses `fs.stat` instead
- [x] add `{ async: true }` option to helper methods to optionally make them asynchronous (they will return promises if this option is set)
  - **why?** because we now use [AVA](/sindresorhus/ava) (which is concurrent) in some projects and having an async option for these helpers will help to improve test performance
- [x] stability: make `is_empty` tests run on CI, teardown, etc

### CODE REVIEW

File: lib/test_helpers.coffee [hunk 1]
```diff
@@ -20,62 +20,132 @@ class Helpers
...
+
+      matches_dir: (dir, expected, opts = { async: false }) ->
         dir = _path(dir)
         expected = _path(expected)
-        String(fs.readdirSync(dir)) == String(fs.readdirSync(expected))
+        if opts.async
+          W.all([nodefn.call(fs.readdir, dir), nodefn.call(fs.readdir, expected)])
+            .spread (dir, expected) -> String(dir) == String(expected)
+        else
+          String(fs.readdirSync(dir)) == String(fs.readdirSync(expected))
```
Comment (hhsnopek) [inline lib/test_helpers.coffee]: Considering you're creating type `String`, you should just use the `===` as it won't perform type conversions

Comment (declandewet) [inline lib/test_helpers.coffee]: `coffee-script` does this automatically: http://coffeescript.org/#try:alert%20'500'%20%3D%3D%20500

Comment (hhsnopek) [inline lib/test_helpers.coffee]: oh, didn't realize it interchanged `==` and `===`, been looking at javascript for too long hah

**Comment by hhsnopek** [comment]
Looks good to me, you cleared up the only thought I had on the matter :)

**Comment by jenius** [comment]
This is great, an excellent set of changes, well executed, and nice clean commits. On board with all the changes here as well. Gold PR star :star2: 

I'll have this shipped out in a moment
